src/utils/api.js

```diff
@@ -3,8 +3,20 @@ export const api = async ({ url, method, body }) => {
     method: method,
     body: JSON.stringify(body),
   });
-
   const data = await response.json();
 
   return data;
 };
+
+export const apiCallWithStatusCode = async ({ url, method, body }) => {
```

Comment:
u can modify the existing api call instead of making a new one, or u can also put a try catch on the api call and throw an error if goes into the catch

Reply:
throwing here has the same problem as the comment above, I just made api use apiCallWithStatusCode